src/jwerl.erl

```diff
@@ -42,9 +47,9 @@ sign(Data, Algorithm) ->
 % Token = jwerl:sign(#{key =&gt; &lt;&lt;"Hello World"&gt;&gt;}, hs256, &lt;&lt;"s3cr3t k3y"&gt;&gt;).
 % </pre>
 % @end
--spec sign(Data :: map() | list(), Algorithm :: algorithm(), KeyOrPem :: binary()) -> binary().
-sign(Data, Algorithm, KeyOrPem) when (is_map(Data) orelse is_list(Data)), is_atom(Algorithm), is_binary(KeyOrPem) ->
-    encode(jsx:encode(Data), config_headers(#{alg => algorithm_to_binary(Algorithm)}), KeyOrPem).
+-spec sign(Data :: map() | list(), Algorithm :: algorithm(), KeyOrPem :: binary(), Opts :: map()) -> binary().
```

Comment:
Options must be `map() | list()`. Indeed, In Elixir we prefer the proplists syntax and (maybe) the map syntax in Erlang ;)

Reply:
Copy that. But I saw some line using `maps:get/2` to read `Opts` only?  e.g: `jwerl#L117`

Is it a bug?  Or Elixir will convert these data types before call `verify/sign` function?